### PR TITLE
JBPM-8153 Container is removed from UI even though it was not possibl…

### PR DESCRIPTION
**Pull request by bxf12315**
…e to stop it.

**Thank you for submitting this pull request**

**JIRA**: [JBPM-8153](https://issues.redhat.com/browse/JBPM-8153)

**Referenced Pull Requests**:
* paste the link(s) from GitHub here
* link 2
* link 3 etc.

**Business Central**: [WAR file](https://donwnload.here/something)

**VS Code**: [plugin](https://donwnload.here/something)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
jenkins do cdb

**Comment by bxf12315** [comment]
jenkins do cdb

**Comment by bxf12315** [comment]
jenkins do fdb

**Comment by bxf12315** [comment]
jenkins do fdb

**Comment by bxf12315** [comment]
jenkins retest this

**Comment by bxf12315** [comment]
jenkins retest this

**Comment by bxf12315** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java [hunk 1]
```diff
@@ -52,11 +55,23 @@ public boolean isRunningContainer(ContainerSpec containerSpec) {
             KieServicesClient client = kieServerInstanceManager.getClient(serverInstanceKey.getUrl());
 
             QueryServicesClient queryServicesClient = client.getServicesClient(QueryServicesClient.class);
-            List<ProcessInstance> processInstances = queryServicesClient.findProcessInstancesByContainerId(containerSpec.getId(), Arrays.asList(0, 1, 4), 0, 100);
+            Collection<ProcessDefinition> processDefinitions = null;
+            try {
+                processDefinitions = queryServicesClient.findProcessesByContainerId(containerSpec.getId(), 0, 10);
+            } catch (Exception e) {
+                if (e.getMessage().contains("org.jbpm.services.api.DeploymentNotFoundException")) {
```
Comment (sutaakar) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java:62]: I suppose that you are trying to check here if the container id exists.
Could you rather leverage KieServicesClient methods for this (either getContainerInfo or listContainers)?

**Comment by pefernan** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java (hunk 1 above)
Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java:62]: ok, But they have the same logic, they all will throw an exception.

Comment (sutaakar) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java:62]: Using findProcessesByContainerId seems to me like misusing the method to retrieve unrelated information.

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java:62]: So, I think we cannot use the way, the container always exits when it's an empty project.

**Comment by pefernan** [comment]
jenkins do fdb

**Comment by pefernan** [comment]
jenkins do fdb

**Comment by pefernan** [comment]
jenkins retest this please

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java (hunk 1 above)
Comment (sutaakar) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java:62]: Why don't you just use just the `findProcessInstancesByContainerId` below?
You are interested just in process instances, not process definitions, right?

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
jenkins do fdb

**Comment by bxf12315** [comment]
jenkins retest this

**Comment by bxf12315** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java (hunk 1 above)
Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/ContainerServiceImpl.java:62]: I think the bug root reason is empty project do not have process definition, so if call findProcessInstancesByContainerId, will thrwo exception. so we need to detect it's process definition, if it does not have one, we can close it.

**Comment by bxf12315** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenter.java [hunk 2]
```diff
@@ -234,9 +236,17 @@ private void updateView(final Collection<Container> containers) {
...
-                return false;
+        if (containerSpec.getStatus().equals(KieContainerStatus.STOPPED)) {
+            for (final Container container : containers) {
+                if (!container.getStatus().equals(KieContainerStatus.FAILED) && !container.getStatus().equals(KieContainerStatus.STOPPED)) {
+                    return false;
+                }
+            }
+        } else {
+            for (final Container container : containers) {
+                if (!container.getStatus().equals(KieContainerStatus.STOPPED)) {
```
Comment (pefernan) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenter.java:247]: @bxf12315 just a question why we are not checking here if the status is failed?

**Pull request closed without merging**